test/tools/unified-spec-runner/match.ts

```diff
@@ -523,6 +567,51 @@ export function matchesEvents(
   }
 }
 
+export function compareLogs(
+  expected: ExpectedLogMessage[],
+  actual: ExpectedLogMessage[],
+  entities: EntitiesMap
+): void {
+  expect(actual).to.have.lengthOf(expected.length);
+
+  for (const [index, actualLog] of actual.entries()) {
+    const rootPrefix = `expectLogMessages[${index}]`;
+    const expectedLog = expected[index];
+
+    // Check that log levels match
+    expect(actualLog).to.have.deep.property('level', expectedLog.level);
+
+    // Check that components match
+    expect(actualLog).to.have.deep.property('component', expectedLog.component);
+
+    // NOTE: The spec states that if the failureIsRedacted flag is present, we
+    // must assert that a failure occurred.
+    if (expectedLog.failureIsRedacted !== undefined) {
+      if (expectedLog.failureIsRedacted) {
+        // Assert that a failure is present and has been redacted
+        expect(actualLog.data.failure, 'Can only use failureIsRedacted when a failure exists').to
+          .exist;
+        expect(actualLog.data.failure, 'Expected failure to have been redacted').to.deep.equal({});
+      } else {
+        // Assert that a failure is present and has not been redacted
+        expect(actualLog.data.failure, 'Can only use failureIsRedacted when a failure exists').to
```

Comment:
This is unnecessarily repetitive, just move the assertion up to just below L589, since it's independent of the actual value of the flag